src/components/ButtonList/ButtonList.js

```diff
@@ -1,19 +1,44 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const ButtonList = props => {
-	return (
-		<div>
-			{props.items.map(item => (
-				<div key={item.id}>
-					<button onClick={() => props.onClick(item.id, item.name)}>
-						{item.name}
-					</button>
-				</div>
-			))}
-		</div>
-	);
-};
+class ButtonList extends React.Component {
+	state = {
+		currentList: this.props.items,
+		userEntry: ''
+	};
+
+	componentDidUpdate(prevProps, prevState) {
+		if (prevState.userEntry !== this.state.userEntry) {
+			const currentList = this.props.items.filter(item =>
+				item.name.includes(this.state.userEntry)
```

Comment:
There's a new API for deriving state from props (getDerivedStateFromProps), though in this case I don't think we need to store it in state. The less state the better!

currentList can be derived from userEntry (as you are doing above) so I'd just have a method returning this and using it in render. If there were performance concerns of doing this we could memoise the function to only create a new list when the userEntry had changed

e.g.

```
getFilteredList() {
  return this.props.items.filter(item => item.name.includes(this.state.userEntry))
}

render() {
  const filteredList = this.getFilteredList();

  ...
}
```